src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -4554,4 +4552,47 @@ public void archiveVersion(Long id) {
             }
         }
     }
+    
+    public boolean isSameTermsOfUseForAllFiles() {
+        if (sameTermsOfUseForAllFiles != null) {
+            return sameTermsOfUseForAllFiles;
+        }
+        if (workingVersion.getFileMetadatas().isEmpty()) {
+            sameTermsOfUseForAllFiles = true;
+            return sameTermsOfUseForAllFiles;
+        }
+        TermsOfUse firstTermsOfUse = workingVersion.getFileMetadatas().get(0).getTermsOfUse();
+        
+        for (FileMetadata fileMetadata : workingVersion.getFileMetadatas()) {
+            if (!isSameTermsOfUse(firstTermsOfUse, fileMetadata.getTermsOfUse())) {
+                sameTermsOfUseForAllFiles = false;
+                return sameTermsOfUseForAllFiles;
+            }
+        }
+        
+        sameTermsOfUseForAllFiles = true;
+        return sameTermsOfUseForAllFiles;
+    }
+    
+    public TermsOfUse getTermsOfUseOfFirstFile() {
+        if (workingVersion.getFileMetadatas().isEmpty()) {
+            return null;
```

Comment:
I'm not a fan of returning null's, we should return Empty object or use Optional if we know it can be null here.

Reply:
ok